test/unit/lib/whitehall/asset_manager_storage_test.rb

```diff
@@ -265,8 +232,48 @@ class Whitehall::AssetManagerStorage::FileTest < ActiveSupport::TestCase
     asset_path = "path/to/ässet.png"
     file = Whitehall::AssetManagerStorage::File.new(asset_path)
 
-    Plek.stubs(:new).returns(stub("plek", asset_root: "http://assets-host"))
-
     assert_equal "http://assets-host/government/uploads/path/to/%C3%A4sset.png", file.url
   end
+
+  test "constructs the url of the file using the assets root and legacy url path" do
+    expected_asset_url = URI.join("http://assets-host", @asset_url_path).to_s
+
+    assert_equal expected_asset_url, @file.url
+  end
+
+  test "returns file url using asset_manager_id when the model has the original asset" do
+    model = build(:attachment_data_with_asset)
+    model.use_non_legacy_endpoints = true
+    model.save!
```

Comment:
I reckon this save! is unnecessary. The code in the `url` method does a database lookup when it doesn't have to.

Reply:
As discused, carrierwave magickery makes it so that `build(:attachment_data_with_asset)` causes the `file` instance to be an instance of `Carrierwave::File`. Whereas if we do save and reload, the file instance ends up being `AssetManagerStorage::File`. This is sad and makes save and reload necessary for these tests.